### PR TITLE
Introduce Concourse pipeline

### PR DESCRIPTION
**Pull request by Everlag**
This pipeline performs two functions.

First, it performs an integration test of the
upgrade-google-client branch whenever new changes appear there.

Second, it runs integration tests against every PR with the
`integrate` label. This allows whitelisted PRs to be tested
automatically before being merged. Status is updated on the
PR when the test completes.

When #210 lands, switch the pipeline to work against master.

### CODE REVIEW

File: ci/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+trust the provided certificate.
+
+* If logging into a specific team, ie `fog-google`, use `--team-name fog-google` to specify that.
+
+## Dockerfile
+
+The [docker-image](https://github.com/fog/fog-google/blob/master/ci/docker-image) directory contains
+the Dockerfile necessary for recreating the Docker image used in tasks. This is referenced
+in a per-task basis as `image_resource`, ie in
+image_resource[run-int.yml](https://github.com/fog/fog-google/blob/master/ci/tasks/run-int.yml)
```
Comment (icco) [inline ci/README.md]: add a space between `image_resource[`

File: ci/credentials.yml.tpl [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# Google Cloud Platform project to run under
+google_project:
+# Google Compute Engine Service Account email
+google_client_email:
+# Google Compute Engine Service Account JSON
+google_json_key_data: |
+
+# An access token with repo:status access
+github_access_token:
+# A SSH private key associated with the access token's account
```
Comment (icco) [inline ci/credentials.yml.tpl]: Mention it can also be a repo deploy push key.

File: ci/tasks/run-int.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+platform: linux
+image_resource:
+  type: docker-image
+  source: {repository: everlag/fog-google}
```
Comment (icco) [inline ci/tasks/run-int.yml]: s/everlag/fog

File: ci/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      access_token: {{github_access_token}}
+      private_key: {{github_private_key}}
+      label: integrate
+      repo: fog/fog-google
+
+resource_types:
+  - name: pull-request
+    type: docker-image
+    source:
+      repository: jtarchie/pr
```
Comment (Everlag) [inline ci/pipeline.yml]: This a third party docker resource, it provides its own docker image.

https://github.com/jtarchie/pullrequest-resource

File: ci/README.md (hunk 1 above)
Comment (Everlag) [inline ci/README.md]: Done

File: ci/credentials.yml.tpl (hunk 2 above)
Comment (Everlag) [inline ci/credentials.yml.tpl]: Done

File: ci/tasks/run-int.yml (hunk 3 above)
Comment (Everlag) [inline ci/tasks/run-int.yml]: This refers to an uploaded docker image created using ci/docker-image/Dockerfile, can you confirm that `fog/fog-google` at dockerhub is populated with that image?

Comment (icco) [inline ci/tasks/run-int.yml]: Can we just have this build the docker file each time?

File: ci/pipeline.yml (hunk 4 above)
Comment (icco) [inline ci/pipeline.yml]: Ah cool, could you link to that in a comment in this file?

File: ci/tasks/run-int.yml (hunk 3 above)
Comment (Everlag) [inline ci/tasks/run-int.yml]: Concourse doesn't support that. A core assumption is that docker images live somewhere outside concourse. You can build docker images using concourse, they need to be uploaded externally before being used internally.

Comment (icco) [inline ci/tasks/run-int.yml]: Lame. Alright.

Comment (Everlag) [inline ci/tasks/run-int.yml]: How would you like to proceed?

I'm happy to keep `everlag/fog-google` available but it'd be better if maintainers had direct access to update the image in the future. To build and upload the image to an account should be a two command process once docker is installed and authenticated.

Comment (icco) [inline ci/tasks/run-int.yml]: Let's just leave it now, and I'll move it to fog in the future.

File: ci/pipeline.yml (hunk 4 above)
Comment (Everlag) [inline ci/pipeline.yml]: Done. I missed this reply earlier, sorry about that.

**Pull request merged**